source/logic.rst

```diff
@@ -0,0 +1,1223 @@
+``λ n, s n + t n`` converges to ``a + b``.
+It is helpful to have a clear pen-and-paper
+proof in mind before you start writing a formal one.
+Given ``ε`` greater than ``0``,
+the idea is to use the hypotheses to obtain an ``Ns``
+such that beyond that point, ``s`` is within ``ε / 2``
+of ``a``,
+and an ``Nt`` such that beyond that point, ``t`` is within
+``ε / 2`` of ``b``.
+Then, whenever ``n`` is greater than or equal to the
+maximum of ``Ns`` and ``Nt``,
+the sequence ``λ n, s n + t n`` should be within ``ε``
+of ``a + b``.
+The following example begins to implement this strategy.
+See if you can finish it off.
+
+.. code-block:: lean
+
+    import data.real.basic
+
+    def converges_to (s : ℕ → ℝ) (a : ℝ) :=
+    ∀ ε, 0 < ε → ∃ N, ∀ n, N ≤ n → abs (s n - a) < ε
+
+    -- BEGIN
+    variables {s t : ℕ → ℝ} {a b : ℝ}
+
+    theorem converges_to_add
+      (cs : converges_to s a) (ct : converges_to t b):
+    converges_to (λ n, s n + t n) (a + b) :=
+    begin
+      intros ε εpos, dsimp,
+      have ε2pos : 0 < ε / 2,
+      { linarith },
+      cases cs (ε / 2) ε2pos with Ns hs,
+      cases ct (ε / 2) ε2pos with Nt ht,
+      use max Ns Nt,
+      sorry
+    end
+    -- END
+
+For hints, you can use ``le_of_max_le_left`` and ``le_of_max_le_right``,
+and ``norm_num`` can prove ``ε / 2 + ε / 2 = ε``.
+Also, it is helpful to use the ``congr`` tactic to
+show that ``abs (s n + t n - (a + b))`` is equal to
+``abs ((s n - a) + (t n - b)),``
+since then you can use the triangle inequality.
+Notice that we marked all the variables ``s``, ``t``, ``a``, and ``b``
+implicit because they can be inferred from the hypotheses.
+
+Proving the same theorem with multiplication in place
+of addition is tricky.
+We will get there by proving some auxiliary statements first.
+See if you can also finish off the next proof,
+which shows that if ``s`` converges to ``a``,
+then ``λ n, c * s n`` converges to ``c * a``.
+It is helpful to split into cases depending on whether ``c``
+is equal to zero or not.
+Since we have not shown you how to do that yet,
+we have taken care of the zero case first,
+and we have left you to prove the result with
+the extra assumption that ``c`` is nonzero.
+
+.. code-block:: lean
+
+    import data.real.basic
+
+    def converges_to (s : ℕ → ℝ) (a : ℝ) :=
+    ∀ ε, 0 < ε → ∃ N, ∀ n, N ≤ n → abs (s n - a) < ε
+
+    theorem converges_to_const (a : ℝ) : converges_to (λ x : ℕ, a) a :=
+    sorry
+
+    variables {s : ℕ → ℝ} {a : ℝ}
+
+    -- BEGIN
+    theorem converges_to_mul_const
+      {c : ℝ} (cs : converges_to s a) :
+    converges_to (λ n, c * s n) (c * a) :=
+    begin
+      by_cases h : c = 0,
+      { convert converges_to_const 0,
+        { ext, rw [h, zero_mul] },
+        rw [h, zero_mul] },
+      have acpos : 0 < abs c,
+        from abs_pos_of_ne_zero h,
+      sorry
+    end
+    -- END
+
+The next theorem is also independently interesting:
+it shows that a convergent sequence is eventually bounded
+in absolute value.
```

Comment:
You should warn readers that this stupid sounding lemma is suboptimal but sufficient for your purposes.